modules/organization/iam.tf

```diff
@@ -32,13 +52,27 @@ locals {
   }
 }
 
+# we use a different key for custom roles to allow referring to the role alias
+# in Terraform, while still being able to define unique role names
+
+check "custom_roles" {
+  assert {
+    condition = (
+      length(local.custom_roles) == length({
+        for k, v in local.custom_roles : v.name => null
+      })
```

Comment:
nit: I'd be explicit and use `distinct` instead of putting the names in a map: `distinct([for _, v in local.custom_roles: v.name])`

Reply:
well, I want to fail as the key clash might imply different values, and generate issues when the role is differently implemented